src/main/java/org/opentripplanner/updater/stoptime/TimetableSnapshotSource.java

```diff
@@ -163,22 +166,30 @@ public void applyTripUpdates(
     // Acquire lock on buffer
     bufferLock.lock();
 
+    Map<TripDescriptor.ScheduleRelationship, Integer> failuresByRelationship = new HashMap<>();
+
     try {
       if (fullDataset) {
         // Remove all updates from the buffer
         buffer.clear(feedId);
       }
 
       LOG.debug("message contains {} trip updates", updates.size());
+      int successfullyApplied = 0;
       int uIndex = 0;
       for (TripUpdate tripUpdate : updates) {
         if (fuzzyTripMatcher != null && tripUpdate.hasTrip()) {
           final TripDescriptor trip = fuzzyTripMatcher.match(feedId, tripUpdate.getTrip());
           tripUpdate = tripUpdate.toBuilder().setTrip(trip).build();
         }
 
+        var tripId = Optional
+          .ofNullable(tripUpdate.getTrip())
+          .map(TripDescriptor::getTripId)
```

Comment:
Should we add the feedId to this already?

Reply:
I've converted a few calls to use the feed-scoped id.

Reply:
I noticed that I now get this a bit hard to interpret error occasionally
```
11:40:04.699 ERROR (GraphUpdaterManager.java:138) Error while running graph writer org.opentripplanner.updater.stoptime.TripUpdateGraphWriterRunnable:
java.lang.IllegalArgumentException: Value can not be null, empty or just whitespace: ''
	at org.opentripplanner.util.AssertUtils.assertHasValue(AssertUtils.java:17)
	at org.opentripplanner.model.FeedScopedId.<init>(FeedScopedId.java:33)
	at org.opentripplanner.updater.stoptime.TimetableSnapshotSource.lambda$applyTripUpdates$0(TimetableSnapshotSource.java:189)
	at java.base/java.util.Optional.map(Optional.java:260)
	at org.opentripplanner.updater.stoptime.TimetableSnapshotSource.applyTripUpdates(TimetableSnapshotSource.java:189)
	at org.opentripplanner.updater.stoptime.TripUpdateGraphWriterRunnable.run(TripUpdateGraphWriterRunnable.java:59)
	at org.opentripplanner.updater.GraphUpdaterManager.lambda$execute$1(GraphUpdaterManager.java:136)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

I think previously this issue was catched with the following log message (this log message still exists in code but I think it will never be used now as this issue is catched earlier):
```
11:44:07.581 WARN (TimetableSnapshotSource.java:201) No trip id found for gtfs-rt trip update: 
trip {
  start_time: "07:00:00"
  start_date: "20220421"
  schedule_relationship: SCHEDULED
  route_id: "6919"
  direction_id: 0
}
stop_time_update {
  arrival {
    time: 1650526275
    uncertainty: 0
  }
  departure {
    time: 1650530640
  }
  stop_id: "6100200"
  schedule_relationship: SCHEDULED
}
stop_time_update {
  arrival {
    time: 1650546840
  }
  departure {
    time: 1650546840
  }
  stop_id: "6110232"
  schedule_relationship: SCHEDULED
}
timestamp: 1650530640

```

Reply:
Ah, a trip ID can also be an empty string! I'm now also handling this case. Can you try again?

Reply:
The same issue still persists
```
12:37:06.389 ERROR (GraphUpdaterManager.java:138) Error while running graph writer org.opentripplanner.updater.stoptime.TripUpdateGraphWriterRunnable:
java.lang.IllegalArgumentException: Value can not be null, empty or just whitespace: ''
	at org.opentripplanner.util.AssertUtils.assertHasValue(AssertUtils.java:17)
	at org.opentripplanner.model.FeedScopedId.<init>(FeedScopedId.java:33)
	at org.opentripplanner.updater.stoptime.TimetableSnapshotSource.lambda$getFeedScopedId$1(TimetableSnapshotSource.java:596)
	at java.base/java.util.Optional.map(Optional.java:260)
	at org.opentripplanner.updater.stoptime.TimetableSnapshotSource.getFeedScopedId(TimetableSnapshotSource.java:596)
	at org.opentripplanner.updater.stoptime.TimetableSnapshotSource.applyTripUpdates(TimetableSnapshotSource.java:186)
	at org.opentripplanner.updater.stoptime.TripUpdateGraphWriterRunnable.run(TripUpdateGraphWriterRunnable.java:59)
	at org.opentripplanner.updater.GraphUpdaterManager.lambda$execute$1(GraphUpdaterManager.java:136)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

Reply:
I made an embarrassing mistake and fixed it here: https://github.com/leonardehrenfried/OpenTripPlanner/commit/b8dbe2d2fba3ad1d4830a2dc702e14748d723ed8

:facepalm: